workspaces/ballerina/ballerina-extension/src/utils/auth-utils.ts

```diff
@@ -0,0 +1,155 @@
+/**
+ * Copyright (c) 2025, WSO2 LLC. (https://www.wso2.com) All Rights Reserved.
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import * as fs from 'fs';
+import * as path from 'path';
+import * as os from 'os';
+
+/**
+ * Checks if we're running in a cloud editor environment
+ * @returns boolean - true if CLOUD_STS_TOKEN environment variable exists
+ */
+export function isCloudEditorEnvironment(): boolean {
+    return !!process.env.CLOUD_STS_TOKEN;
+}
+
+/**
+ * Reads the current access token from Settings.toml
+ * @returns Promise<string | undefined> - The current access token or undefined if not found
+ */
+export async function getCurrentAccessToken(): Promise<string | undefined> {
+    try {
+        const settingsPath = path.join(os.homedir(), '.ballerina', 'Settings.toml');
+        
+        if (!fs.existsSync(settingsPath)) {
+            return undefined;
+        }
+
+        const content = fs.readFileSync(settingsPath, 'utf8');
+        const tokenMatch = content.match(/accesstoken\s*=\s*"([^"]*)"/);
+        
+        return tokenMatch ? tokenMatch[1] : undefined;
+    } catch (error) {
+        console.error('Failed to read current access token:', error);
+        return undefined;
+    }
+}
+
+/**
+ * Checks if a JWT token is expired or about to expire (within 5 minutes)
+ * @param token - The JWT token to check
+ * @returns boolean - true if token is expired or about to expire
+ */
+export function isTokenExpiredOrExpiring(token: string): boolean {
+    try {
+        // JWT tokens have 3 parts separated by dots
+        const parts = token.split('.');
+        if (parts.length !== 3) {
+            return true; // Not a valid JWT format
+        }
+
+        // Decode the payload (second part)
+        const payload = JSON.parse(atob(parts[1]));
+        
+        if (!payload.exp) {
+            return true; // No expiration time means we should refresh
+        }
+
+        // Check if token expires within 5 minutes (300 seconds)
+        const currentTime = Math.floor(Date.now() / 1000);
+        const expirationTime = payload.exp;
+        const bufferTime = 300; // 5 minutes buffer
+        
+        return (expirationTime - currentTime) <= bufferTime;
```

Comment:
will this handle the scenario of the token already expired ?

IMO we can avoid doing this check and always update the settings.toml if we are in code server